src/js/components/MainNav/MainNav.js

```diff
@@ -1,30 +1,95 @@
 import React, { useContext } from 'react';
-import { withRouter } from 'react-router';
 import '../../../scss/components/MainNav/MainNav.scss';
 import { GlobalContext } from '../../contexts/GlobalContextWrapper';
 import AddServerButton from '../AddServerButton/AddServerButton';
 import MainNavButton from '../MainNavButton/MainNavButton';
 import ServerLink from '../ServerLink/ServerLink';
+import useDiscordData from '../../hooks/useDiscordData';
 
-/**
- * Let's wrap the component with withRouter() to make it receptive to route changes
- */
-const MainNav = withRouter(() => (
-  <div className="nav-column nav-column-server">
-    <div className="nav-group">
-      <MainNavButtons />
+const MainNav = () => {
+  const { Config } = useContext(GlobalContext);
+
+  const config = Config.get([
+    'discordAPIResources',
+    'mainNavButtons',
+    'mainNavButtonTypes',
+    'discordUrls',
+  ]);
+
+  const { discordAPIResources, mainNavButtons, mainNavButtonTypes, discordUrls } = config;
+  const { link, addServerButton } = mainNavButtonTypes;
+  const { client, guilds, icons } = discordAPIResources;
+  const { appCDN } = discordUrls;
+
+  const guildData = useDiscordData(guilds, client);
+
+  let addGuildData;
+
+  if (guildData) {
+    // Sort guilds in alphabetical order
+    guildData.sort((a, b) => {
+      const nameA = a.name.toUpperCase();
+      const nameB = b.name.toUpperCase();
+      if (nameA < nameB) {
+        return -1;
+      }
+      if (nameA > nameB) {
+        return 1;
+      }
+      return 0;
```

Comment:
```suggestion
      return nameA < nameB ? 1 : nameA > nameB ? -1 : 0;
```

Reply:
ESLint: Do not nest ternary expressions.(no-nested-ternary)